src/cpp/session/modules/SessionCopilot.cpp

```diff
@@ -985,6 +990,12 @@ void synchronize()
    s_copilotEnabled = isCopilotEnabled();
    s_copilotIndexingEnabled = s_copilotEnabled && isCopilotIndexingEnabled();
    
+   // Subscribe to file monitor if enabled
+   if (s_copilotIndexingEnabled)
+   {
+      static bool once = subscribeToFileMonitor();
```

Comment:
This makes use of the fact that, in calls of the form:

```
static T t = f();
```

The function `f()` will only be evaluated only once, on first use.